stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -648,27 +653,423 @@ extension Unsafe${Mutable}BufferPointer {
+  /// the buffer's `startIndex`. If `source` contains an equal or greater number
+  /// of elements than the buffer can hold, the returned index is equal to
+  /// the buffer's `endIndex`.
   ///
-  /// Warning: All buffer elements must be initialized before calling this. 
-  /// Assigning to part of the buffer must be done using the `assign(repeating:count:)`
-  /// method on the buffer’s `baseAddress`. 
+  /// - Parameter source: A sequence of elements with which to initialize the
+  ///   buffer.
+  /// - Returns: An iterator to any elements of `source` that didn't fit in the
+  ///   buffer, and an index to the next uninitialized element in the buffer.
   @inlinable // unsafe-performance
-  public func assign(repeating repeatedValue: Element) {
+  @_silgen_name("$sSr10initialize4from8IteratorQyd___Sitqd___t7ElementQyd__RszSTRd__lF")
+  public func initialize<S: Sequence>(
+    from source: S
+  ) -> (unwritten: S.Iterator, index: Index) where S.Element == Element {
+    return source._copyContents(initializing: self)
+  }
+
+  /// Initializes the buffer's memory with
+  /// every element of the source.
+  ///
+  /// Prior to calling the `initialize(fromContentsOf:)` method on a buffer,
+  /// the memory referenced by the buffer must be uninitialized,
+  /// or the `Element` type must be a trivial type. After the call,
+  /// the memory referenced by the buffer up to, but not including,
+  /// the returned index is initialized.
+  /// The buffer must reference enough memory to accommodate
+  /// `source.count` elements.
+  ///
+  /// The returned index is the position of the next uninitialized element
+  /// in the buffer, one past the index of the last element written.
+  /// If `source` contains no elements, the returned index is equal to the
+  /// buffer's `startIndex`. If `source` contains as many elements as the buffer
+  /// can hold, the returned index is equal to the buffer's `endIndex`.
+  ///
+  /// - Precondition: `self.count` >= `source.count`
+  ///
+  /// - Note: The memory regions referenced by `source` and this buffer
+  ///     must not overlap.
+  ///
+  /// - Parameter source: A collection of elements to be used to
+  ///     initialize the buffer's storage.
+  /// - Returns: The index one past the last element of the buffer initialized
+  ///     by this function.
+  @inlinable
+  @_alwaysEmitIntoClient
+  public func initialize(
+    fromContentsOf source: some Collection<Element>
+  ) -> Index {
+    let count = source.withContiguousStorageIfAvailable {
+      guard let sourceAddress = $0.baseAddress, !$0.isEmpty else {
+        return 0
+      }
+      _precondition(
+        $0.count <= self.count,
+        "buffer cannot contain every element from source."
+      )
+      baseAddress?.initialize(from: sourceAddress, count: $0.count)
+      return $0.count
+    }
+    if let count {
+      return startIndex.advanced(by: count)
+    }
+
+    var (iterator, copied) = source._copySequenceContents(initializing: self)
```

Comment:
Huh, is there a reason we aren't calling `_copyContents` here?

Reply:
It seemed redundant to have the full `_copyContents` only to double-check its work after it returns. Only one of the stdlib types that implements a `_copyContents` fast path doesn't also have `wCSIA`, and that is going to be fixed soon.